zipline/pipeline/loaders/events.py

```diff
@@ -148,7 +152,8 @@ def __init__(self,
                 raise ValueError(
                     WRONG_MANY_COL_DATA_FORMAT_ERROR.format(sid=k)
                 )
-
+        self.events_by_sid = {sid: df.dropna(subset=[self.event_date_col]) for
```

Comment:
Note: put this back for the time being, since moving the filtering to the blaze loader didn't work (blaze didn't know how to compute the query when I added filtering for nulls to it).